tests/components/homekit/test_type_switches.py

```diff
@@ -449,3 +449,46 @@ async def test_input_select_switch(hass, hk_driver, events, domain):
     assert acc.select_chars["option1"].value is False
     assert acc.select_chars["option2"].value is False
     assert acc.select_chars["option3"].value is False
+
+
+async def test_button_switch(hass, hk_driver, events):
+    """Test switch accessory from a button entity."""
+    domain = "button"
+    entity_id = "button.test"
+
+    hass.states.async_set(entity_id, None)
+    await hass.async_block_till_done()
+    acc = Switch(hass, hk_driver, "Switch", entity_id, 2, None)
+    await acc.run()
+    await hass.async_block_till_done()
+
+    assert acc.activate_only is True
+    assert acc.char_on.value is False
+
+    call_press = async_mock_service(hass, domain, "press")
+
+    await hass.async_add_executor_job(acc.char_on.client_update_value, True)
```

Comment:
```suggestion
    acc.char_on.client_update_value(True)
```

Reply:
HomeKit is fully async now.  These are left over. I've been switching them out when I make changes.